src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * product.div(value1).assertEquals(value2);
+   * product.div(value2).assertEquals(value1);
+   * ```
+   * 
+   * @param value - a `field-like` value to multiply with the {@link Field}.
+   * 
+   * @return A {@link Field} element equivalent to the modular difference of the two value.
    */
-  mul(y: Field | number | string | boolean): Field;
+  mul(value: Field | number | string | boolean): Field;
 
   /**
-   * Divides this {@link Field} element through another coercible to a field.
+   * Divide another `field-like` element through this {@link Field} and returns the result.
+   *
+   * ```ts
+   * const value1 = Field(6);
+   * const value2 = 3; // A `field-like` element
+   * 
+   * const product = value1.div(value2);
+   * 
+   * product.assertEquals(Field(2));
+   * ```
+   * 
+   * **Warning: This is a modular division in the pasta field. You can think it as the reverse operation of modular multiplication.
+   * 
+   * ```ts
+   * const value1 = Field(2);
+   * const value2 = Field(5);
+   * 
+   * const quotient = value1.div(value2);
+   * 
+   * // If you try to print product - `console.log(quotient.toBigInt())` - you will realize that it prints a very big integer, so you cannot use arithmetic operations on snarkyJS as in regular programming languages.
+   * // However, all the operations always stay provable: This means you can use the reverse operation of division (multiplication) to prove the quotient is calculated correctly.
+   * quotient.mul(value2).assertEquals(value1);
+   * ```
+   * 
+   * @param value - a `field-like` value to divide with the {@link Field}.
+   * 
+   * @return A {@link Field} element equivalent to the modular division of the two value.
    */
-  div(y: Field | number | string | boolean): Field;
+  div(value: Field | number | string | boolean): Field;
 
   /**
-   * Squares this {@link Field} element.
+   * Square this {@link Field} element.
    *
    * ```typescript
-   * const x2 = x.square();
-   * x2.assertEquals(x.mul(x));
+   * const someField = Field(7);
+   * const square = someField.square();
+   * 
+   * square.assertEquals(someField.mul(someField)); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * ** Warning: This is a modular multiplication. Please see `mul()` function for more details.
+   * 
+   * @return A {@link Field} element equivalent to the multiplication of the {@link Field} element with itself.
    */
   square(): Field;
 
   /**
-   * Square roots this {@link Field} element.
+   * Take the square root of this {@link Field} element.
    *
    * ```typescript
-   * x.square().sqrt().assertEquals(x);
+   * const someField = Field(42);
+   * const squareRoot = someField.sqrt();
+   * 
+   * squareRoot.mul(squareRoot).assertEquals(someField); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * **Warning: This is a modular square root. Please see `div()` function for more details.
```

Comment:
@yunus433 please fix `**Warning: ..` and `** Warning: ..` and `**Important` without a closing `**` throughout the PR. Also I recommend that you look at these changes in VS code's intellisense to double-check that they look as you intended.

The `**double-star syntax**` is Markdown for **bold text** and will show as bold text in the editor